ZAPD/ZFile.cpp

```diff
@@ -852,9 +852,19 @@ string ZFile::ProcessDeclarations()
 
 	for (pair<uint32_t, Declaration*> item : declarations)
 	{
-		while (item.second->size % 4 != 0)
+		if (item.second->alignment == DeclarationAlignment::Align2)
 		{
-			item.second->size++;
+			while (item.second->size % 2 != 0)
+			{
+				item.second->size++;
+			}
+		}
+		else
+		{	
+			while (item.second->size % 4 != 0)
+			{
+				item.second->size++;
+			}
```

Comment:
Does the current `Alignment` always align the data to `4`, even if the `DeclarationAlignment` is set to `None`?
(This question isn't specific to this PR).

Reply:
I think it does, which is why I had to add the `Align2` value. I would assume that `None` would not cause any alignment, but then I encountered trouble using it for `Vec3s[]`s. This does seem intuitive to me, but I didn't want to change that behavior.

Reply:
Well, this should be fixed in a future PR. I'll add a new issue in the meanwhile.